### DISCUSSION TITLE
Use /whoami to protect backend service

### DISCUSSION THREAD

**Discussion by linfengOu** [ory/kratos#1398] [Q&A] [answered]
Hello,

I recently came across ory/kratos, and would like to adapt it into my go application, but got a blocker:
Say I have a frontend app, exactly same as the one from quickstart. It works well from login/register perspective. But after login, it sends request to an endpoint `/private/something` to my backend go application. I'd like to protect this private endpoint by calling `/whoami` to verify session. But seems kratos go client `ToSession` (whoami) doesn't support session cookie.
From docs it says session token can also be used, but how can I get the session token from one request?

Any suggestion would be appreciated.
Thanks.

**Comment by linfengOu** [comment, marked as answer]
Or should I make http request directly to kratos instance, without using kratos GO client?

**Reply by aeneasr** [reply]
Hm yeah it looks like `openapi-generator` doesn't support the cookie security type. So we've specified it correctly in Ory Kratos' spec (https://github.com/ory/kratos/blob/master/spec/openapi.json#L2002-L2006) but it does not appear to be used in the generated Go code (`kratos-client-go`).

I also looked it if is possible to customize the HTTP headers for just one request but that doesn't appear to be the case. So we'll add some convenience function to make this work for backends as well (right now this SDK method works for frontends)

**Reply by linfengOu** [reply]
Thanks. Glad to hear that you'll take this into consider.

**Reply by tgaddair** [reply]
Is there an issue to track this proposed change? This would be very useful for my application as well.

**Reply by mooijtech** [reply]
Any update on this (it would be nice if kratos-client-go supported this)?

**Reply by mooijtech** [reply]
Ended up calling the /sessions/whoami endpoint from Go myself.

**Reply by aeneasr** [reply]
Yes, see here: https://www.ory.sh/docs/guides/protect-page-login/go
